lxd/instance/drivers/driver_qemu.go

```diff
@@ -1680,6 +1680,18 @@ func (d *qemu) start(stateful bool, op *operationlock.InstanceOperation) error {
 	// onStop hook isn't triggered prematurely (as this function's reverter will clean up on failure to start).
 	monitor.SetOnDisconnectEvent(false)
 
+	// Apply CPU pinning.
+	if cpuInfo.vcpus == nil {
+		if d.architectureSupportsCPUHotplug() && cpuInfo.cores > 1 {
+			err := d.setCPUs(cpuInfo.cores, false)
```

Comment:
We should check how setting limits.cpu back to 1 will at boot time and update time will affect core scheduling setup.

Reply:
Perhaps we need to move the `cpuInfo.cores > 1 ` check into `setCPUs`.

Reply:
Thanks for suggestion! Have reworked this part.

I have added some comments in the code which can be useful in the future.

Reply:
>We should check how setting limits.cpu back to 1 will at boot time and update time will affect core scheduling setup.

Couldn't you elaborate on that? I don't understand the point

Reply:
> Ah, ok. I think I've got it. Your point is that we probably need to refactor `setCPUs` a bit to be more self-contained and cover edge cases like removing all the vCPUs except one.

Thats right, I'm seeing there's a condition for calling setCPUs during start that >1 cores is set, but i dont see that same condition in the live update one. It appears to be some inconsistency there.

Reply:
Have rechecked this again, we don't really want to move `cpuInfo.cores > 1` inside `setCPUs` because the idea behind having this check is to figure out if an instance has more than one cpu (then hotplug is needed). At the same time, `setCPUs` can be called from `limits.cpu` option handler and it's perfectly find to have it called when `cpuInfo.cores == 1`.